src/Raven.Server/Documents/Queries/FieldsToFetch.cs

```diff
@@ -68,8 +69,10 @@ public FieldsToFetch(IndexQueryServerSide query, IndexDefinitionBaseServerSide i
             var mustExtractFromIndex = projectionBehavior.FromIndexOnly();
             var maybeExtractFromIndex = projectionBehavior.FromIndexOrDefault();
             
-            //Since we always store field in Corax we can try to get it from index entry when performing projection. This is an alternative value for `canExtractFromIndex`.  
-            var isCorax = searchEngineType is SearchEngineType.Corax;
+            // Since we always store fields in Corax, we can try to retrieve them from the index entry when performing projection. This is an alternative value for `canExtractFromIndex`.
+            // Additionally, when the index is a map-reduce, we have to skip the fields' stored values and look them up in the stored JSON of the reduce-map.
+            // The reason behind this is related to the type of value we return to the client. In the field, we can have a LazyStringValue, but the client may want a numeric value.
+            var isCorax = searchEngineType is SearchEngineType.Corax && indexType.IsMapReduce() is false;
```

Comment:
1. `isCorax` is no longer a valid name for this variable. Maybe `forceExtractFromIndex`?